packages/type-utils/src/isTypeBrandedLiteralLike.ts

```diff
@@ -0,0 +1,55 @@
+import * as tsutils from 'ts-api-utils';
+import * as ts from 'typescript';
+
+function isLiteralOrTaggablePrimitiveLike(type: ts.Type): boolean {
+  if (tsutils.isUnionOrIntersectionType(type)) {
+    return type.types.every(isLiteralOrTaggablePrimitiveLike);
+  }
+
+  return (
+    type.isLiteral() ||
+    tsutils.isTypeFlagSet(
+      type,
+      ts.TypeFlags.BigInt | ts.TypeFlags.Number | ts.TypeFlags.String,
+    )
+  );
+}
+
+function isObjectLiteralLike(type: ts.Type): boolean {
+  if (tsutils.isUnionOrIntersectionType(type)) {
+    return type.types.every(isObjectLiteralLike);
+  }
```

Comment:
Test: I don't think this part is covered by a test (will be easier to notice once codecov's reports are merged).